crates/polars-time/src/series/mod.rs

```diff
@@ -86,9 +88,32 @@ pub trait TemporalMethods: AsSeries {
         let s = self.as_series();
         match s.dtype() {
             #[cfg(feature = "dtype-date")]
-            DataType::Date => s.date().map(|ca| ca.weekday()),
+            DataType::Date => s.date().map(|ca| {
+                // Closed formula to find weekday, no need to go via Chrono.
+                // The 4 comes from the fact that 1970-01-01 was a Thursday.
+                // We do an extra `+ 7` then `% 7` to ensure the result is non-negative.
+                unary_elementwise_values(ca, |t| (((t - 4) % 7 + 7) % 7 + 1) as i8)
+            }),
             #[cfg(feature = "dtype-datetime")]
-            DataType::Datetime(_, _) => s.datetime().map(|ca| ca.weekday()),
+            DataType::Datetime(time_unit, time_zone) => s.datetime().map(|ca| {
+                match time_zone.as_deref() {
+                    Some("UTC") | None => {
+                        // fastpath!
+                        // Same idea as above, but we need to subtract 1 for dates
+                        // before 1970-01-01 with non-zero sub-daily components.
+                        let divisor = match time_unit {
+                            TimeUnit::Milliseconds => MILLISECONDS_IN_DAY,
+                            TimeUnit::Microseconds => MICROSECONDS_IN_DAY,
+                            TimeUnit::Nanoseconds => NANOSECONDS_IN_DAY,
+                        };
+                        unary_elementwise_values(ca, |t| {
+                            let t = t / divisor - ((t < 0 && t % divisor != 0) as i64);
+                            (((t - 4) % 7 + 7) % 7 + 1) as i8
```

Comment:
Aside: I suspect this is incorrect in UTC due to the presence of leap seconds: you can occasionally have a fencepost error where, if the date is during a leap second, the computed day is on the wrong side. I don't know if `chrono` also had this issue.

Reply:
thanks @wence- for taking a look

this shouldn't be an issue https://docs.rs/chrono/latest/chrono/naive/struct.NaiveTime.html#leap-second-handling

> Chrono does not try to accurately implement leap seconds; it is impossible. Rather, it allows for leap seconds but behaves as if there are no other leap seconds. Various operations will ignore any possible leap second(s) except when any of the operands were actually leap seconds.

Do you have an example where it is?